frontend/src/component/pages/Home.tsx

```diff
@@ -93,6 +94,11 @@ export class Home extends Component<Props, State> {
       }
       this.setState(newState);
     });
+    const longLink = this.getLongLinkFromQueryParams();
+    if (validateLongLinkFormat(longLink) == null) {
+      this.props.store.dispatch(updateLongLink(longLink));
+      this.shortLinkTextField.current!.focus();
+    }
```

Comment:
I recommend extracting this into a function with meaningful name so that it's easier for reader to under this feature!